bin/ocamlformat-rpc/main.ml

```diff
@@ -18,6 +18,26 @@ Caml.at_exit (Format.pp_print_flush Format.err_formatter);;
 
 Caml.at_exit (Format_.pp_print_flush Format_.err_formatter)
 
+module IO = struct
+  type 'a t = 'a
+
+  type ic = In_channel.t
+
+  type oc = Out_channel.t
+
+  let ( >>= ) x f = f x
+
+  let return x = x
+
+  let read_line ic = In_channel.input_line ic
```

Comment:
This primitive is a little suspicious. What if there's a new line inside the Csexp.t?